build/olm-catalog/yaml-options-gen.py

```diff
@@ -0,0 +1,432 @@
+            self.default = ('' if option['default'] == 'None'
+                            else int(float(option['default'])))
+
+            # If we have min and max defined, add it to the help
+            if '(' in option['type']:
+                self.help += ' ' + option['type'][option['type'].index('('):]
+
+        # TODO: Implement support for these 3 types in the operator
+        elif option['type'] == 'Dict of String':
+            self.form_type = None
+            self.ignore = True
+            self.default = option['default']
+            sys.stderr.write('Ignoring %s with type %s\n' % (option['name'],
+                                                             option['type']))
+        elif option['type'] == 'List of String':
+            self.form_type = None
+            self.ignore = True
+            self.default = option['default']
+            sys.stderr.write('Ignoring %s with type %s\n' % (option['name'],
+                                                             option['type']))
+        elif option['type'] == 'List of IPAddress':
+            self.form_type = None
+            self.ignore = True
+            self.default = option['default']
+            sys.stderr.write('Ignoring %s with type %s\n' % (option['name'],
+                                                             option['type']))
+
+        else:
+            raise Exception('Unkown type %s' % option['type'])
+
+        if option['required'] == 'True':
+            self.help += ' [Optional]'
+
+    def add_driver(self, driver):
+        self.drivers.add(driver)
+
+
+def include_driver(drivername):
+    if EXCLUDE_DRIVERS is not None:
+        for regex in EXCLUDE_DRIVERS:
+            if re.match(regex, drivername, re.IGNORECASE):
+                return False
+
+    if INCLUDE_DRIVERS is None:
+        return True
+
+    for regex in INCLUDE_DRIVERS:
+        if re.match(regex, drivername, re.IGNORECASE):
+            return True
+
+    return False
+
+
+def render_option(option, driver, group_options=False):
+    if option.ignore:
+        return ''
+    group = driver + "Options" if group_options else 'DriverSettings'
+
+    result = """
+        - description: {description}
+          displayName: {display_name}
+          path: config.envVars.X_CSI_BACKEND_CONFIG.{name}
+          x-descriptors:
+            - 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:{group}'
+            - {opt_type}
+            - 'urn:alm:descriptor:com.tectonic.ui:fieldDependency:config.envVars.X_CSI_BACKEND_CONFIG.driver:{driver}'
+    """.format(name=option.name,  # noqa
+               display_name=option.display_name,
+               description='"' + option.help + '"',
+               opt_type=option.form_type,
+               driver=driver,
+               group=group)
+
+    # The only way to present the groups folded is to set them as advanced
+    if group_options:
+        result += 12 * ' ' + "- 'urn:alm:descriptor:com.tectonic.ui:advanced'"
```

Comment:
I think these should be 8 instead of 12

Reply:
Yup, it's clear [in the output](https://github.com/embercsi/ember-csi-operator/blob/6029712b36a0454f4d464a3c5b000c0b9cddbdd3/deploy/olm-catalog/next/ember-csi-operator.vX.Y.Z.clusterserviceversion.yaml#L3048):

```
            - 'urn:alm:descriptor:com.tectonic.ui:fieldDependency:config.envVars.X_CSI_BACKEND_CONFIG.driver:AS13000'
                - 'urn:alm:descriptor:com.tectonic.ui:advanced'
```